### PR TITLE
clean up ddc temp dir on exit

### PR DESCRIPTION
**Pull request by jakemac53**

### CODE REVIEW

File: lib/src/barback/dartdevc/dartdevc_environment.dart [hunk 1]
```diff
@@ -41,8 +42,16 @@ class DartDevcEnvironment {
     _scratchSpace = new ScratchSpace(_getAsset);
   }
 
-  /// Deletes the [_scratchSpace].
-  Future cleanup() => _scratchSpace.delete();
+  /// Deletes the [_scratchSpace] and shuts down the workers.
+  Future cleanup() {
+    return Future.wait([
```
Comment (natebosch) [inline lib/src/barback/dartdevc/dartdevc_environment.dart]: [optional] => instead of return?

File: lib/src/command/serve.dart [hunk 2]
```diff
@@ -99,6 +100,17 @@ class ServeCommand extends BarbackCommand {
         basePort: port,
         compiler: compiler,
         environmentConstants: environmentConstants);
+
+    var sigintListener;
```
Comment (natebosch) [inline lib/src/command/serve.dart]: can we give a type here?

File: lib/src/barback/asset_environment.dart [hunk 3]
```diff
@@ -184,6 +184,11 @@ class AssetEnvironment {
       this._hostname, this._basePort, this.environmentConstants, this.compiler,
       {this.dartDevcEnvironment});
 
+  /// Performs any necessary cleanup before shutdown.
+  Future cleanup() async {
```
Comment (munificent) [inline lib/src/barback/asset_environment.dart]: `cleanUp()`.

File: lib/src/command/serve.dart [hunk 4]
```diff
@@ -99,6 +100,17 @@ class ServeCommand extends BarbackCommand {
...
+    sigintListener = ProcessSignal.SIGINT.watch().listen((_) {
+      sigintListener.cancel();
+      log.progress('Cleaning up dartdevc temp environment..', () async {
```
Comment (munificent) [inline lib/src/command/serve.dart]: Ellipses aren't needed. `progress()` will handle that for you.

File: lib/src/barback/asset_environment.dart (hunk 3 above)
Comment (jakemac53) [inline lib/src/barback/asset_environment.dart]: done

File: lib/src/command/serve.dart (hunk 4 above)
Comment (jakemac53) [inline lib/src/command/serve.dart]: done

File: lib/src/command/serve.dart (hunk 2 above)
Comment (jakemac53) [inline lib/src/command/serve.dart]: done

**Pull request merged**